### PR TITLE
Feat: Video skeleton

### PR DESCRIPTION
**Pull request by muselesscreator**
image

Moves header and footer to per-editor components.
creates new video redux reducer
Created new widget components for each form field, with default values, and a set/get hook each.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -69,7 +70,8 @@
     "redux-thunk": "^2.4.1",
     "reselect": "^4.1.5",
     "tinymce": "^5.10.2",
-    "lodash-es": "^4.17.21"
+    "video-react": "^0.15.0",
+    "video.js": "^7.18.1"
```
Comment (connorhaugh) [inline package.json:74]: Pre sync questions: makes sense we are adding both now, before we become stuck to one, but we should run through the pros and cons of each.

File: src/editors/containers/TextEditor/components/EditorFooter/__snapshots__/index.test.jsx.snap [hunk 2]
```diff
@@ -2,10 +2,10 @@
 
 exports[`EditorFooter snapshots Save Failed, error message raised 1`] = `
 <div
-  className="editor-footer fixed-bottom"
```
Comment (connorhaugh) [inline src/editors/containers/TextEditor/components/EditorFooter/__snapshots__/index.test.jsx.snap:5]: That is a merge conflict, plz rectify (fixed-bottom is the bootstrap rule we are using)

File: src/editors/containers/TextEditor/components/EditorFooter/index.jsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+export const EditorFooter = ({
+  editorRef,
+  // injected
+  intl,
+}) => {
+  const dispatch = useDispatch();
+  const isInitialized = hooks.isInitialized();
+  const saveFailed = hooks.saveFailed();
+  return (
+    <div className="editor-footer mt-auto">
```
Comment (connorhaugh) [inline src/editors/containers/TextEditor/components/EditorFooter/index.jsx:26]: This is really where I meant to make the mt-auto vs fixed: bottom note.

File: src/editors/containers/TextEditor/components/EditorFooter/__snapshots__/index.test.jsx.snap [hunk 4]
```diff
@@ -60,20 +53,14 @@ exports[`EditorFooter snapshots Save Failed, error message raised 1`] = `
 
 exports[`EditorFooter snapshots not intialized, Spinner appears and button is disabled 1`] = `
 <div
-  className="editor-footer fixed-bottom"
+  className="editor-footer mt-auto"
```
Comment (muselesscreator) [inline src/editors/containers/TextEditor/components/EditorFooter/__snapshots__/index.test.jsx.snap:56]:
```suggestion
  className="editor-footer fixed-bottom"
```

File: src/editors/containers/TextEditor/components/EditorFooter/index.jsx (hunk 3 above)
Comment (muselesscreator) [inline src/editors/containers/TextEditor/components/EditorFooter/index.jsx:26]:
```suggestion
    <div className="editor-footer fixed-bottom">
```

**Comment by rayzhou-bit** [comment]
Tried this out locally and seems good to me. Will have to test out upload again when it reaches stage. I assume the two lint errors we can ignore as we will have to change this during video editor work?

File: src/editors/containers/TextEditor/hooks.js [hunk 5]
```diff
@@ -1,115 +1,54 @@
...
-    toolbar: module.getConfig('toolbar'),
-    imagetools_toolbar: module.getConfig('imageToolbar'),
-    imagetools_cors_hosts: [lmsEndpointUrl], // as image assets come from lms, we need to whitelist it.
-    height: '100%',
-    min_height: 500,
-    branding: false,
+    plugins: pluginConfig.plugins,
+    imagetools_toolbar: pluginConfig.imageToolbar,
+    toolbar: pluginConfig.toolbar,
+    contextmenu: 'link table',
```
Comment (connorhaugh) [inline src/editors/containers/TextEditor/hooks.js:47]: Is this the change that makes it so we don't get the right click menu on images?

File: src/editors/containers/TextEditor/hooks.test.jsx [hunk 6]
```diff
@@ -1,187 +1,218 @@
...
-        );
-        Object.values(module.pluginConfig.imageToolbar).forEach(
-          value => expect(output.init.imagetools_toolbar.includes(value)).toBe(true),
+        expect(output.init.plugins).toEqual(pluginConfig.plugins);
+        expect(output.init.imagetools_toolbar).toEqual(pluginConfig.imageToolbar);
+        expect(output.init.toolbar).toEqual(pluginConfig.toolbar);
+        Object.keys(pluginConfig.config).forEach(key => {
+          expect(output.init[key]).toEqual(pluginConfig.config[key]);
+        });
+        //expect(output.init.imagetools_cors_hosts).toMatchObject([props.lmsEndpointUrl]);
```
Comment (connorhaugh) [inline src/editors/containers/TextEditor/hooks.test.jsx:115]: probably should put this test back in, should be a quick fix.

**Review by connorhaugh — COMMENTED** [review]
Looks good to me upon further review, just would recommend to knock out those two lint fixes and add back in the one test for the cors_host in tinymce config. Then we can get this merged in and ready to go!

File: src/editors/containers/TextEditor/hooks.js (hunk 5 above)
Comment (muselesscreator) [reply src/editors/containers/TextEditor/hooks.js:47]: yep

File: src/editors/containers/TextEditor/hooks.test.jsx (hunk 6 above)
Comment (muselesscreator) [reply src/editors/containers/TextEditor/hooks.test.jsx:115]: I fully pulled out the cors hosts and it still worked, which is why I commented out the test.  will add back in if we decide we need it for the image tools?

**Comment by connorhaugh** [comment]
Yeah checked it in devstack. Several Key Issues:
-All tinyMCE modals (grid, html editor, etc) are hidden under the EditorContainer Modal. Please set the Z index of that modal to be 1 or 0 for now, so it doesn't get in the way. 
- One I will note with comments close on code, which leads to an error saving.

File: src/editors/containers/TextEditor/index.jsx [hunk 7]
```diff
@@ -47,58 +43,63 @@ export const TextEditor = ({
...
   return (
-    <div className="editor-body h-75 overflow-auto">
-      <ImageUploadModal
-        isOpen={isOpen}
-        close={closeModal}
-        editorRef={editorRef}
-        {...imageSelection}
-      />
+    <EditorContainer
+      getContent={hooks.getContent({ editorRef })}
```
[the remark this answers is not in the document]
Comment (muselesscreator) [reply src/editors/containers/TextEditor/index.jsx:54]: yes!

**Review by connorhaugh — APPROVED** [review]
Looks Good! Still want to do some exercising of everything when this gets on stage (which I can do), but seems a worthwhile improvement! Thanks for taking my refactoring considerations into account, this should help people a lot in future.

**Pull request merged**